#! /usr/bin/env python

PACKAGE="deep_vision_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#        name        type     level    description       default  min  max
gen.add("box_threshold", double_t, 0, "Detection Box threshold", 0.5, 0.0, 1.0)
gen.add("text_threshold", double_t, 0, "Detection Text threshold", 0.5, 0.0, 1.0)
gen.add("nms_threshold", double_t, 0, "Detection NMS threshold", 0.8, 0.0, 1.0)
gen.add("classes",   str_t, 0, "Classes to grounded detect. You can split classes by ;", "person; keyboard; monitor")
exit(gen.generate(PACKAGE, "deep_vision_ros", "GroundingDINO"))
